File: ADN-cursos-back/java-arquitectura-hexagonal/microservicio/dominio/src/main/java/com/ceiba/clase/modelo/entidad/Clase.java
package com.ceiba.clase.modelo.entidad;

import lombok.Getter;

import static com.ceiba.dominio.ValidadorArgumento.validarObligatorio;

@Getter
public class Clase {
    private static final String SE_DEBE_INGRESAR_EL_CURSO = "Se debe ingresar el curso";
    private static final String SE_DEBE_INGRESAR_EL_DOCENTE = "Se debe ingresar el docente";
    private static final String SE_DEBE_INGRESAR_EL_NOMBRE_DE_LA_CLASE = "Se debe ingresar el nombre de la clase";
    private Long id;
    private String nombre;
    private Long docente;
    private Long curso;

    public Clase(Long id, String nombre, Long docente, Long curso){
        validarObligatorio(nombre, SE_DEBE_INGRESAR_EL_NOMBRE_DE_LA_CLASE);
        validarObligatorio(docente, SE_DEBE_INGRESAR_EL_DOCENTE);
        validarObligatorio(curso, SE_DEBE_INGRESAR_EL_CURSO);
        this.id = id;
        this.nombre = nombre;
        this.docente = docente;
        this.curso = curso;
    }

}

File: ADN-cursos-front/angular-base/src/app/feature/clase/shared/model/clase.ts
export class Clase {
    id?                 : number;
    nombre              : string;
    docente             : number;
    curso               : number;
    nombreDocente?      : string;
    nombreCurso?        : string;
    
    constructor(id: number, nombre: string, docente: number, curso: number){
        this.id         = id;
        this.nombre     = nombre;
        this.docente    = docente;
        this.curso      = curso;
    }
}